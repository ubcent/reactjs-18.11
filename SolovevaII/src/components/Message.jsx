import React, {Component} from 'react';

export class Message extends Component{
    render() {

    }

}