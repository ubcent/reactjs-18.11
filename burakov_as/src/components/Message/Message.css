.message {
  background-color: lightblue;
  padding: 5px 15px;
  border-radius: 20px;
  font-size: 20px;
  margin: 5px;
}

.message-owner {
  align-self: flex-end;
}

.message-companion {
  align-self: flex-start;
}

.message-sender {
  color: gray;
  font-size: 14px;
}
