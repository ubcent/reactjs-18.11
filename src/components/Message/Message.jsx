import './Message.scss';

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

export const messageType =  {
  author: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};

export class Message extends Component {
  static propTypes = messageType;

  render() {
    const { author, text } = this.props;

    const classes = classNames('message', {
      'message-owner': author !== 'Bot',
      'message-companion': author === 'Bot',
    });

    return (
      <div className={classes}>
        <div>{text}</div>
        <div className="message-sender">{author}</div>
      </div>
    );
  }
}