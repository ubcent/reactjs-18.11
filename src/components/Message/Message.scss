.message {
    background-color:lightblue;
    padding: 5px 15px;
    border-radius: 20px;
    font-size: 20px;
    margin: 5px;

    &-owner {
        align-self: flex-end;
    }

    &-companion {
        align-self: flex-start;
    }

    &-sender {
        color: gray;
        font-size: 14px;
    }
}