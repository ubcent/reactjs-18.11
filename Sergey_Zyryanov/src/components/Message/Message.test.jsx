import React from 'react';
import {Message} from './Message';

import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

test('Message should be rendered', () => {

  const tree = renderer
    .create(<Message author={'Serg'} text={'hello'}/>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('Message should be rendered as div', () => {
  const message = shallow(<Message author={'Serg'} text={'hello'}/>);
  const div = message.find('div');

  expect(div.length).toBe(3);
  expect(div.first().text()).toBe('Serg');
  expect(div.last().text()).toBe('hello');
});