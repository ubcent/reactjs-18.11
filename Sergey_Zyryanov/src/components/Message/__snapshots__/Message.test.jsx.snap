// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should be rendered 1`] = `
<div
  className="message message-owner"
>
  <div>
    hello
  </div>
  <div
    className="message-sender"
  >
    Serg
  </div>
</div>
`;
